apps/openmw/mwgui/inventorywindow.cpp

```diff
@@ -528,14 +525,26 @@ namespace MWGui
         if (mDragAndDrop->mIsOnDragAndDrop)
         {
             MWWorld::Ptr ptr = mDragAndDrop->mItem.mBase;
+            int itemType = ptr.getContainerStore()->getType(ptr);
+
             mDragAndDrop->finish();
 
             if (mDragAndDrop->mSourceModel != mTradeModel)
             {
                 // Move item to the player's inventory
                 ptr = mDragAndDrop->mSourceModel->moveItem(mDragAndDrop->mItem, mDragAndDrop->mDraggedCount, mTradeModel);
             }
+
             useItem(ptr);
+
+            // If item is ingredient or potion don't stop drag and drop to simplify action of taking more than one 1 item
+            if ((itemType == MWWorld::ContainerStore::Type_Ingredient
+                    || itemType == MWWorld::ContainerStore::Type_Potion)
+                && mDragAndDrop->mDraggedCount > 1)
+            {
+                dragItem (nullptr, mDragAndDrop->mDraggedCount - 1);
```

Comment:
Here it is assumed that mSelectedItem is the item being dragged. I don't think that is the case if the drag started from an external window, i.e. pick an item from a container directly onto the character portrait.